src/ILCompiler.Compiler/src/Compiler/Compilation.cs

```diff
@@ -118,6 +120,9 @@ public bool HasLazyStaticConstructor(TypeDesc type)
 
         public MethodDebugInformation GetDebugInfo(MethodIL methodIL)
         {
+            if (!GenerateDebugInfo)
```

Comment:
> What's the scenario for not generating debug info?

"Optimizing" and "generating debug info" are two separate things on most compilers (Roslyn, gcc, VC++, etc.), controlled by separate compiler switches (that are off by default). This was also requested by the JIT team (see the issue in the pull request).